pandas/core/reshape/merge.py

```diff
@@ -1441,9 +1441,13 @@ def _factorize_keys(lk, rk, sort=True):
         rk = rk.values
 
     # if we exactly match in categories, allow us to use codes
+    if isinstance(lk, CategoricalIndex):
+        ldata = lk._data
+    else:
+        ldata = lk
```

Comment:
instead of this, you can implement ``is_dtype_equal`` on a CI, its pretty easy, something like

```
def is_dtype_equal(self, other):
    return self._data.is_dtype_equal(other)
```